warehouse/query-core/src/main/java/datawave/query/index/lookup/EntryParser.java

```diff
@@ -73,7 +80,7 @@ public EntryParser(ASTEQNode node, String fieldName, String literal, Set<String>
             }
         }
         
-        if (!skipNodeDelay && Union.isDay(date) && !Union.isShard(date) && info.uids().isEmpty()) {
```

Comment:
The code for Union.isDay() and Union.isShard() is identical. The IsShard() method is just the isDay() method with a ! before it. We can safely remove it.